### ISSUE TITLE
Support Particle Functionalities

### ISSUE THREAD

**Issue by cindytsai** [calab-ntu/libyt#32]
## Support Particle Functionalities
Initially, particle plots may generate false figure, if the MPI size is too large. After testing on different machine, this doesn't seems to be an issue in `libyt`. This is more or less related to memory space of one machine has. But we still cannot find where actually is this issue, we move to another issue.
- #41 

> :information_source: After testing on calab912 and eureka, it doesn't seem to be an issue. Instead, it is more or less related to memory on the machine. Though we still don't know why and where is the bug.

> :heavy_check_mark: Particle plot (`ParticlePlot`, `ParticleProjectionPlot`) can successfully run in parallel in inline-analysis. Though some memory related issues may occurred. 
---
#### Test Run on My Laptop with RAM = 16
All of the images are results from inline-analysis. `MPI=1` is the correct and expected outcome.
- Test Problem: `gamer` Plummer
- Machine: My Laptop
##### ParticlePlot
```python
import yt
yt.enable_parallelism()
def yt_inline():
    ds = yt.frontends.libyt.libytDataset()
    par = yt.ParticlePlot( ds, 'particle_position_x', 'particle_position_y', "particle_mass", center='c' )
    if yt.is_root():
        par.save()
```
* MPI=1
  ![Fig000000000_Particle_z_particle_mass](https://user-images.githubusercontent.com/6188366/138090870-95af7cad-6f94-4799-8696-dffec2841f31.png)
* MPI=2
  ![Fig000000000_Particle_z_particle_mass](https://user-images.githubusercontent.com/6188366/138090937-fb8b0151-0be5-45c1-9f54-664f08e2ee10.png)

##### ParticleProjectionPlot
```python
import yt
yt.enable_parallelism()
def yt_inline():
    ds = yt.frontends.libyt.libytDataset()
    par = yt.ParticleProjectionPlot( ds, "z")
    if yt.is_root():
        par.save()
```
* MPI=1
  ![Fig000000000_Particle_z_particle_ones](https://user-images.githubusercontent.com/6188366/138091632-19fe2308-2b79-40b8-b8ed-349c26b77e23.png)
* MPI=2
  ![Fig000000000_Particle_z_particle_ones](https://user-images.githubusercontent.com/6188366/138091715-af1caa0d-9761-4d30-892e-04ef82a6a631.png)
 * The upper-right cluster is different.

**Comment by cindytsai** [comment]
## ParticlePlot
- Test Problem: `gamer` Plummer
- Machine: calab912
- Inline Script
```python
import yt
yt.enable_parallelism()
def yt_inline():
    ds = yt.frontends.libyt.libytDataset()
    par = yt.ParticlePlot( ds, 'particle_position_x', 'particle_position_y', "particle_mass", center='c' )
    if yt.is_root():
        par.save()
```
- Output: For MPI = 1, 2, 3, 4, they all generate the same figure.
  ![Fig000000001_Particle_z_particle_mass](https://user-images.githubusercontent.com/6188366/141424137-4691b895-45bf-460c-9627-d6e14019c5c4.png)

**Comment by cindytsai** [comment]
## ParticlePlot
- Test Problem: `gamer` Plummer
- Machine: eureka
- MPI Version: 3.1.5-intel-oneapi
- Inline Script:
  ```python
  ```
- Run Command:
  ```bash
  OMPI_MCA_osc=sm,pt2pt mpirun -np 4 --output-filename log ./gamer
  ```
  - Should put `OMPI_MCA_osc=sm.pt2pt` at the front, otherwise will reach an assertion error when encountered RMA operations.
- Output:

**Comment by cindytsai** [comment]
## ParticleProjectionPlot
- Test Problem: `gamer` Plummer
- Machine: eureka
- MPI Version: 3.1.5-intel-oneapi
- Inline Script:
  ```python
  import yt
  yt.enable_parallelism()
  def yt_inline():
      ds = yt.frontends.libyt.libytDataset()
      par_prj = yt.ParticleProjectionPlot( ds, "z" )
      if yt.is_root():
          par_prj.save()
  ```
- Run Command:
  ```bash
  OMPI_MCA_osc=sm,pt2pt mpirun -np 32 --output-filename log ./gamer
  ```
